app/connectors/parsers/v2/GetPenaltyDetailsParser.scala

```diff
@@ -0,0 +1,71 @@
+ * limitations under the License.
+ */
+
+package connectors.parsers.v2
+
+import models.v2.GetPenaltyDetails
+import play.api.http.Status._
+import play.api.libs.json.{JsError, JsSuccess}
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+import utils.Logger.logger
+
+object GetPenaltyDetailsParser {
+  sealed trait GetPenaltyDetailsFailure
+  sealed trait GetPenaltyDetailsSuccess
+
+  case class  GetPenaltyDetailsSuccessResponse(penaltyDetails: GetPenaltyDetails) extends GetPenaltyDetailsSuccess
+  case class  GetPenaltyDetailsFailureResponse(status: Int) extends GetPenaltyDetailsFailure
+  case object GetPenaltyDetailsMalformed extends GetPenaltyDetailsFailure
+
+  type GetPenaltyDetailsResponse = Either[GetPenaltyDetailsFailure, GetPenaltyDetailsSuccess]
+
+  implicit object GetPenaltyDetailsReads extends HttpReads[GetPenaltyDetailsResponse] {
+    override def read(method: String, url: String, response: HttpResponse): GetPenaltyDetailsResponse = {
+      response.status match {
+        case OK =>
+          logger.debug(s"[GetPenaltyDetailsReads][read] Json response: ${response.json}")
+          response.json.validate[GetPenaltyDetails] match {
+            case JsSuccess(getPenaltyDetails, _) =>
+              Right(GetPenaltyDetailsSuccessResponse(getPenaltyDetails))
+            case JsError(errors) =>
+              logger.debug(s"[GetPenaltyDetailsReads][read] Json validation errors: $errors")
+              Left(GetPenaltyDetailsMalformed)
+          }
+        case BAD_REQUEST =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Bad Request when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(BAD_REQUEST))
+        case NOT_FOUND =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Received Not Found when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(NOT_FOUND))
+        case CONFLICT =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Received Conflict when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(CONFLICT))
+        case UNPROCESSABLE_ENTITY =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Received Unprocessable Enity when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(UNPROCESSABLE_ENTITY))
+        case INTERNAL_SERVER_ERROR =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Received ISE when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(INTERNAL_SERVER_ERROR))
+        case SERVICE_UNAVAILABLE =>
+          logger.error(s"[GetPenaltyDetailsReads][read] Received Service Unavailable when trying to call GetPenaltyDetails - with body: ${response.body}")
+          Left(GetPenaltyDetailsFailureResponse(SERVICE_UNAVAILABLE))
+        case _@status =>
```

Comment:
This code seems to be repeated quite a lot, instead we could do:
```suggestion
        case status@(NOT_FOUND | BAD_REQUEST | CONFLICT | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE) => {
          logger.error(s"[GetPenaltyDetailsReads][read] Received $status when trying to call GetPenaltyDetails - with body: ${response.body}")
          Left(GetPenaltyDetailsFailureResponse(status))
        }
        case _@status =>
```